middleware.ts

```diff
@@ -0,0 +1,4 @@
+// eslint-disable-next-line no-restricted-exports, import/no-default-export
+export { default } from "next-auth/middleware"
+
+export const config = { matcher: ["/", "/assets/:path*", "/checkout/:path*", "/search"] }
```

Comment:
in a perfect world, we'd be able to tap into the middleware and if the user isn't authorized, redirect them to `/auth/signin?origin={route}`. do you think we'd be able to that at all?

Reply:
I'm not sure if I see why you would prefer that pattern? If we don't want every page to be secured, we would need some white/black-listing anyway. If not (the way it actually is now) I would very much prefer that the [documented way of protecting all pages](https://next-auth.js.org/tutorials/securing-pages-and-api-routes#nextjs-middleware) just worked, but it doesn't 🤷‍♀️ 

But I guess it is possible by using the `unstable_getServerSession` to check for an active session in the middleware and redirect if not? 🤔

Reply:
does the current solution redirect the user to the page they were on before getting the sign in page?

Reply:
i guess it's not a huge deal if it doesn't

Reply:
I does, due to the last commit. Men nekter egentlig å belite meg på at det skal være nødvendig 👀